pymc/distributions/timeseries.py

```diff
@@ -243,142 +266,289 @@ def logp(
-    init: distribution
-        distribution for initial values (Defaults to Flat())
+    rho: tensor_like of float
+        Tensor of autoregressive coefficients. The n-th entry in the last dimension is
+        the coefficient for the n-th lag.
+    sigma: tensor_like of float, optional
+        Standard deviation of innovation (sigma > 0). Defaults to 1. Only required if
+        tau is not specified.
+    tau: tensor_like of float
+        Precision of innovation (tau > 0).
+    constant: bool, optional
+        Whether the first element of rho should be used as a constant term in the AR
+        process. Defaults to False
+    init_dist: distribution, optional
+        Distribution for initial values. Defaults to Normal(0, sigma). Distribution
+        should be  created via the `.dist()` API, and have  dimension (*size, ar_order).
+        If not, it will be automatically resized.
+    ar_order: int, optional
+        Order of the AR process. Inferred from length of the last dimension of rho, if
+        possible. ar_order = rho.shape[-1] if constant else rho.shape[-1] - 1
+
+    Notes
+    -----
+    The init distribution will be cloned, rendering it distinct from the one passed as
+    input.
+
+    Examples
+    --------
+    .. code-block:: python
+
+        # Create an AR of order 3, with a constant term
+        with pm.Model() as AR3:
+            coefs = pm.Normal("coefs", 0, size=4)
+            init = pm.Normal.dist(5, size=3)
+            ar3 = pm.AR("ar3", coefs, sigma=1.0, init_dist=init, constant=True, steps=500)
+
     """
 
-    def __init__(self, rho, sigma=None, tau=None, constant=False, init=None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        tau, sigma = get_tau_sigma(tau=tau, sigma=sigma)
-        self.sigma = at.as_tensor_variable(sigma)
-        self.tau = at.as_tensor_variable(tau)
+    @classmethod
+    def dist(
+        cls,
+        rho,
+        sigma=None,
+        tau=None,
+        *,
+        init_dist=None,
+        steps=None,
+        constant=False,
+        ar_order=None,
+        **kwargs,
+    ):
+        _, sigma = get_tau_sigma(tau=tau, sigma=sigma)
+        sigma = at.as_tensor_variable(floatX(sigma))
+        rhos = at.atleast_1d(at.as_tensor_variable(floatX(rho)))
 
-        self.mean = at.as_tensor_variable(0.0)
+        if "init" in kwargs:
+            warnings.warn(
+                "init parameter is now called init_dist. Using init will raise an error in a future release.",
+                FutureWarning,
+            )
+            init_dist = kwargs["init"]
+
+        steps = get_steps_from_shape(steps, kwargs.get("shape", None))
+        steps = at.as_tensor_variable(intX(steps), ndim=0)
+
+        if ar_order is None:
+            # TODO: Test case where this fails and passing ar_order directly solves it
+            # We need to know the order of the AR. We perform constant_folding here
+            # to retrieve them
+            shape_fg = FunctionGraph(
```

Comment:
Maybe add a comment as to what's happening here, hard to see.

Reply:
Done